docs/learn.md

```diff
@@ -126,10 +126,281 @@ OCaml. Some examples are the upgrade of the compiler version used by Melange to
+global switches, in the same way we can have a `node_modules` folder local to
+our project or install global dependencies using `yarn global` or `npm install
+-g`. Global switches can be handy sometimes, but to avoid confusion, the
+recommendation is to avoid them.
+
+The default settings can be changed if the `-a` option is omitted while calling
+`opam init`.
+
+#### Minimal `app.opam` file
+
+The equivalent to `package.json` is an `app.opam` file, where `app` is the name
+of the package. It is possible to have multiple opam files in the same directory
+or project.
+
+There is no opam command to manipulate the opam file. A command similar to `npm
+init` or `yarn add` does not exist in opam, so the updates in `.opam` files have
+to be done by hand.
+
+A minimal `.opam` file looks like this:
+
+```
+opam-version: "2.0"
+name: "my-app"
+authors: "Louis"
+homepage: "https://github.com/khady/example"
+maintainer: "ex@ample.com"
+dev-repo: "git+ssh://git@github.com:khady/example.git"
+bug-reports: "https://github.com/khady/example/issues"
+version: "0.1"
+build: [
+  [ "dune" "subst" ] {pinned}
+  [ "dune" "build" "-p" name "-j" jobs ]
+]
+depends: [
+  "dune" {build}
+]
+```
+
+`build:` tells opam that `dune` is needed only to build the project.
+
+#### Installing packages
+
+The first thing we need is a local switch in the current project. To verify if a
+switch exists already, we can look for a `_opam` directory at the root of the
+project or use the `opam switch` command to identify if a switch already exists
+in the project folder.
+
+If it does not exist, we can create it with:
+
+```bash
+opam switch create . 4.14.1 --deps-only
+```
+
+If it exists, we can install the dependencies of the project with:
+
+```bash
+opam install . --deps-only
+```
+
+#### Add new packages
+
+To add a new package to the opam switch, we can do:
+
+```bash
+opam install <package_name>
+```
+
+But opam will not modify the `app.opam` file during the installation, this has
+to be done by hand. It is as simple as adding the name of the package in the
+`depends` field.
+
+#### Linking packages for development
+
+This can be achieved with `opam pin`. For example, to pin a package to a
+specific commit on GitHub:
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#61bfbfaf8c971dec5152bce7e528d30552c70bc5
+```
+
+Branch names can also be used.
+
+```bash
+opam pin add reason-react.dev https://github.com/reasonml/reason-react.git#feature
+```
+
+A shortcut to get the latest version of a package is to use the `--dev-repo`
+flag, e.g.
+
+```bash
+opam pin add reason-react.dev --dev-repo
+```
+
+To remove the pinning for any package, use `opam unpin <package_name>` or `opam
+pin remove <package_name>`.
+
+For other options, the command is well described in [the official
+documentation](https://opam.ocaml.org/doc/Usage.html#opam-pin).
+
+#### Upgrading packages
+
+To upgrade the installed packages to the latest version, run:
```

Comment:
this doesn't do anything if you don't update the opam-repository first with `opam update`.

Reply:
I just realized it's mentioned at the end; but should be mentioned before because people are going to copy paste the command without reading.

Reply:
Switched the order in https://github.com/melange-re/documentation-site/pull/24/commits/5335568db4f55618918920867d1c53bfa971ef98.